### PR TITLE
Ar/restructure2

### PR DESCRIPTION
**Pull request by AlexisRenchon**
## Purpose 

This PR is the same as PR #220 but I recreated a branch from main as I was struggling a bit with git rebase... 
We can delete PR #220 after we merge this PR. 
In this PR, I created `/Standalone` and `/Integrated` folders in `/experiments`, `/src`, and `/test`. 
I also accordingly corrected all paths in `src/ClimaLSM.jl`. 
I believe it should be all set and ready to merge, but it is possible I missed something (a path may be broken, maybe?), we will see if CI passes. 
I also put `src/Regridder.jl` and `src/Parameters.jl` in `src/SharedUtilities`, even if it was already done in PR #245. - I hope that's okay. 
I hope it is ready to merge, let me know! 

## To-do

Just check if CI pass... this is just folder restructure, so it should be straightforward and ready to merge if paths have been corrected properly. 

## Content

Create `/Standalone` and `/Integrated` folder throughout the repository for clarity and organisation. 

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by AlexisRenchon** [comment]
I am still debugging with file path... WIP

**Comment by kmdeck** [comment]
> I am still debugging with file path... WIP
> 
> EDIT: CI passed except formatting, but I am not sure what needs to be done

from ClimaLSM.jl/
"julia .dev/climaformat.jl ."

**Comment by kmdeck** [comment]
There are also some paths in the docs (I think just links within ClimaLSM) that need to be updated: 

for example, 
https://github.com/CliMA/ClimaLSM.jl/blob/f1b75b9819bd08a10e904963dccbee4c21eec151/docs/tutorials/Usage/LSM_single_column_tutorial.jl#L31C46-L31C46

**Comment by kmdeck** [comment]
Possibly a future PR to update the docs structure to reflect this naming structure as well?

**Comment by AlexisRenchon** [comment]
@kmdeck It looks like it is ready to be merged! I just don't know why JuliaFormatter failed? Do I need to do something? 

If you give me your green light, I will squash the commits and merge this PR.

File: docs/src/folderstructure.md [hunk 1]
```diff
@@ -44,9 +44,9 @@ For example, `/experiments/LSM/ozark/` contains:
...
+- standalone: This folder contains standalone models, which are submodels that can be run independently of each other. This is an important aspect of `ClimaLSM` code design: to maximise modularity, sub-models can be run alone, and many different methods of the same sub-model can be defined via Julia multiple-disptach. The standalone folder is independent from the integrated folder. 
+- integrated: This folder contains integrated models. It assembles standalone models together, as one would assemble pieces of a puzzle. Thanks to the modularity of `ClimaLSM` design, many configuration of LSM can be assembled in integrated models. The same functions (`update_aux!`, `exp_tendency!`, etc.) can be used for standalone and integrated models, and an can be stepped  in the same way.
 
 As well as 3 files:
```
Comment (kmdeck) [inline docs/src/folderstructure.md:51]: 1 file

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -39,28 +39,28 @@ steps:
...
       - label: "Evaporation from sand"
-        command: "julia --color=yes --project=experiments experiments/Standalone/Soil/evaporation.jl"
-        artifact_paths: "experiments/Standalone/Soil/evaporation*png"
+        command: "julia --color=yes --project=experiments experiments/standalone/Soil/evaporation.jl"
+        artifact_paths: "experiments/standalone/Soil/evaporation*png"
 
       - label: "ozark_test_hydrology_only"
-        command: "julia --color=yes --project=experiments experiments/LSM/ozark/hydrology_only/ozark.jl"
+        command: "julia --color=yes --project=experiments experiments/integrated/ozark/hydrology_only/ozark.jl"
         artifact_paths: "experiments/LSM/ozark/hydrology_only/*png"
```
Comment (kmdeck) [inline .buildkite/pipeline.yml:59]: LSM -> integrated here (the plots did not upload to buildkite webpage)

File: docs/tutorials/Usage/domain_tutorial.jl [hunk 3]
```diff
@@ -52,7 +52,7 @@ but it is defined for all domains[^1].
 
 There is a single key method which take a ClimaLSM domain as an argument.
 
-- [` coordinates(domain)`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.Domains.coordinates): under the hood, this function  uses
+- [` coordinates(domain)`](https://clima.github.io/ClimaLSM.jl/dev/APIs/sharedutilities/#ClimaLSM.Domains.coordinates): under the hood, this function  uses
```
Comment (kmdeck) [inline docs/tutorials/Usage/domain_tutorial.jl:55]: many of these links are not working in the generated docs... not sure why, not seeing an obvious mistake.

Comment (AlexisRenchon) [reply docs/tutorials/Usage/domain_tutorial.jl:55]: sharedutilities, lowercase

Comment (kmdeck) [reply docs/tutorials/Usage/domain_tutorial.jl:55]: the link itself does work if you type it into the web browser, so somehow just in this PR something changed about the path?

**Comment by kmdeck** [comment]
This is not related to your PR, but i noticed another bug in our docs. Could you change, in sharedutilities.md, these lines as follows?
```
## Drivers
```@docs
ClimaLSM.Drivers.PrescribedAtmosphere
ClimaLSM.Drivers.PrescribedRadiativeFluxes
ClimaLSM.Drivers.AbstractAtmosphericDrivers
ClimaLSM.Drivers.AbstractRadiativeDrivers
ClimaLSM.Drivers.surface_fluxes_at_a_point
ClimaLSM.Drivers.radiative_fluxes_at_a_point
ClimaLSM.Drivers.construct_atmos_ts
ClimaLSM.Drivers.surface_air_density
ClimaLSM.Drivers.liquid_precipitation
ClimaLSM.Drivers.snow_precipitation
```
to
```
## Drivers
```@docs
ClimaLSM.PrescribedAtmosphere
ClimaLSM.PrescribedRadiativeFluxes
ClimaLSM.AbstractAtmosphericDrivers
ClimaLSM.AbstractRadiativeDrivers
ClimaLSM.surface_fluxes_at_a_point
ClimaLSM.radiative_fluxes_at_a_point
ClimaLSM.construct_atmos_ts
ClimaLSM.surface_air_density
ClimaLSM.liquid_precipitation
ClimaLSM.snow_precipitation
```
?

thank you!

**Comment by kmdeck** [comment]
sorry, one more thing! 
in test, I think we could do:
test/run_tests.jl
test/aqua.jl
tests/shared_utilities/implicit_timestepping/
tests/shared_utilities/variable_types.jl
tests/shared_utilities/domains.jl
tests/shared_utilities/utilities.jl
tests/standalone/
tests/integrated/

File: docs/tutorials/Usage/domain_tutorial.jl (hunk 3 above)
Comment (AlexisRenchon) [reply docs/tutorials/Usage/domain_tutorial.jl:55]: maybe! I guess it can be a small fix after we merge this, in case the path is a bit different.

**Comment by AlexisRenchon** [comment]
@kmdeck done! 
Note I also changed sharedutilities to shared_utilities

File: docs/tutorials/Usage/domain_tutorial.jl (hunk 3 above)
Comment (kmdeck) [reply docs/tutorials/Usage/domain_tutorial.jl]: that's fine with me. ill add it to the docs overhaul issue

**Comment by kmdeck** [comment]
> @kmdeck It looks like it is ready to be merged! I just don't know why JuliaFormatter failed? Do I need to do something?
> 
> If you give me your green light, I will squash the commits and merge this PR.

what happens when you do this?
from ClimaLSM.jl/
"julia .dev/climaformat.jl .
git status"

File: docs/tutorials/Usage/domain_tutorial.jl (hunk 3 above)
Comment (AlexisRenchon) [reply docs/tutorials/Usage/domain_tutorial.jl]: I think it was because it is shared_utilities.

**Pull request closed without merging**